### PR TITLE
Djikstra optimized

### PR DESCRIPTION
**Pull request by vishnoitanuj**
As of errors stated in #560 . The issue related solved with code optimization

### CODE REVIEW

File: DataStructures/Graphs/BFS.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			}
+		}
+    }
+    
+    private static void printBFS(int edges[][],int sv,boolean visited[]){
+        Queue<Integer> q=new LinkedList<>();
+        q.add(sv);
+        int n=edges.length;
+        visited[sv]=true;
+        String s=new String();
```
Comment (Amanda1232) [inline DataStructures/Graphs/BFS.java:35]: Our static analysis engine detects that your pull request includes a change that **method invokes inefficient new String() constructor**. According to [the best practice](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-constructor-dm-string-void-ctor), creating a new java.lang.String object using the no-argument constructor **wastes memory** because the object so created will be functionally indistinguishable from the empty string constant "".  Java guarantees that identical string constants will be represented by the same String object.  **Therefore, you should just use the empty string constant directly.**

File: DataStructures/Graphs/DFS.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import java.util.*;
+
+public class DFS {
```
Comment (prashantdoshi28) [inline DataStructures/Graphs/DFS.java:3]: Please make separate PRs for unrelated codes.

File: DataStructures/Graphs/BFS.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+
+public class BFSs {
```
Comment (prashantdoshi28) [inline DataStructures/Graphs/BFS.java:3]: Please make separate PRs for unrelated codes.

File: Others/Dijkstra.java [hunk 4]
```diff
@@ -1,192 +1,52 @@
...
+	public static int findMinVertex(int a[],boolean visited[]) {
+		int min=-1;
+		for(int i=0;i<a.length;i++) {
+			if(!visited[i] && (min==-1 || a[i]<a[min]))
+				min=i;
+		}
+		return min;
+	}
+
+	public static void main(String[] args) {
```
Comment (prashantdoshi28) [inline Others/Dijkstra.java:37]: Try not to use this approach of having 
```java
public static void main(String[] args) {...}
```
call your actual method. 

Instead create your class while keeping in mind of how others will use this i.e. by creating objects and calling your actual methods. So make your methods 
```java
public <return-type> methodName()
```

Please make this changes.

**Review by prashantdoshi28 — CHANGES_REQUESTED** [review]
I have requested some changes, please go through my comments.
